.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,12 +1,22 @@
-<!--
-Please follow the naming conventions in the "guidelines for contributing" link above or the pull request may be rejected.
+Fixes #
 
-Also, do consider outlining the solution taken as doing so will likely help in the reviewing process
-(e.g., when the pull request involves non-trivial changes)
--->
+**PR Checklist**
 
-Fixes #
+<!-- Remove this portion after you have made the checks. -->
+
+Have you:
+- [ ] Read and understood [our PR guideline (steps 4-6)](https://github.com/TEAMMATES/teammates/blob/master/docs/process.md#step-4-submit-a-pr)?
+  - [ ] Added the issue number to the "Fixes" keyword above?
+  - [ ] Titled the PR as specified in the abovementioned document?
+- [ ] Made your changes on a branch other than `master`?
+- [ ] Used good commit messages?
```

Comment:
We can dive straight to adopting that guideline for reasons:
- Adopting something takes time. Weeks, months even. The maintainers need to first get used to it before enforcing it to everyone else.
- Stringent commit standard does help for `git blame` and stuff, but we need to make sure that it does not become counter-productive: instead of spending time on the code quality which ultimately translates to software quality, time is spent on organizing commits and reviewing them. (pretty sure this is why many OSS projects enforce a one-commit-per-PR rule, even pre-merging)
- We use squash-and-merge workflow which makes individual commits obsolete. Of course this is precisely the workaround due to the sub-optimal commits, but until we're ready to move to rebase-and-merge workflow, stringent commit standard is not necessary.

Having said them, I think I'll exclude this item from the list for now.